### PR TITLE
feat: disallow `()` after enum case

### PR DESCRIPTION
**Pull request by mlutze**
~This implementation allows the `()` and parses as if it were not there. The other option is to disallow it.~

This implementation disallows `()` at the end of an enum case. (E.g. `case C()`)

fixes https://github.com/flix/flix/issues/3002

### CODE REVIEW

**Comment by magnus-madsen** [comment]
Question: Would it be better to emit a WeederError explaining what to do? (Give more precise errors when possible)

**Comment by mlutze** [comment]
It would make the parsing a little awkward (adding another case only for this particular error) but it's not unmanageable.

File: main/src/ca/uwaterloo/flix/language/phase/Parser.scala [hunk 1]
```diff
@@ -148,7 +148,7 @@ class Parser(val source: Source) extends org.parboiled2.Parser {
         }
 
         def CaseWithType: Rule1[ParsedAst.Case] = namedRule("Case") {
-          SP ~ Names.Tag ~ Type ~ SP ~> ParsedAst.Case
+          SP ~ Names.Tag ~ Types.Tuple ~ SP ~> ParsedAst.Case
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/language/phase/Parser.scala:151]: Does this still allow a record? I guess it does?

Comment (mlutze) [reply main/src/ca/uwaterloo/flix/language/phase/Parser.scala:151]: You can do a record row like `C(())` but that would be ill-kinded anyway.

A proper record `C({})` is still possible.

**Pull request merged**